app/src/main/java/org/xbmc/kore/ui/sections/settings/SettingsFragment.java

```diff
@@ -217,4 +233,76 @@ public boolean onPreferenceClick(Preference preference) {
             }
         });
     }
+
+    private void setupLanguagePreference(final ListPreference languagePref) {
+        Locale[] locales = getLocales(BuildConfig.SUPPORTED_LOCALES);
+
+        Arrays.sort(locales, new Comparator<Locale>() {
+            @Override
+            public int compare(Locale o1, Locale o2) {
+                return o1.getDisplayName().compareToIgnoreCase(o2.getDisplayName());
+            }
+        });
+
+        String[] displayNames = new String[locales.length];
+        String[] entryValues = new String[locales.length];
+        for(int index = 0; index < locales.length; index++) {
+            Locale locale = locales[index];
+            displayNames[index] = locale.getDisplayName(locale);
+            entryValues[index] = getLanguageCountryCode(locale);
+        }
+
+        languagePref.setValue(getLanguageCountryCode(getCurrentLocale()));
+        languagePref.setEntries(displayNames);
+        languagePref.setEntryValues(entryValues);
+        languagePref.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object o) {
+                languagePref.setValue(o.toString());
+                updatePreferredLanguage(o.toString());
+                return true;
+            }
+        });
+    }
+
+    private String getLanguageCountryCode(Locale locale) {
```

Comment:
https://developer.android.com/reference/java/util/Locale.html#toLanguageTag()

in v21+

Reply:
See my comment below.